boot/zephyr/flash_map_extended.c

```diff
@@ -24,15 +24,14 @@ BOOT_LOG_MODULE_DECLARE(mcuboot);
 /* MEMORY MAPPED for XiP on external NOR flash takes the sspi-nor or ospi-nor or qspi-nor device */
 #define FLASH_DEVICE_ID SPI_FLASH_0_ID
 #if DT_NODE_HAS_STATUS(DT_INST(0, st_stm32_xspi_nor), okay)
-#define DT_DRV_COMPAT st_stm32_xspi_nor
 #define FLASH_DEVICE_NODE DT_INST(0, st_stm32_xspi_nor)
-#define FLASH_DEVICE_BASE DT_REG_ADDR_BY_IDX(DT_INST_PARENT(0), 1)
+#define FLASH_DEVICE_BASE DT_REG_ADDR_BY_IDX(DT_PARENT(FLASH_DEVICE_NODE),1)
```

Comment:
The initial line is correct. Instance should actually be taken into account (in case another xspi controller is dealing with another memory as in N6 case.)
This should rather be extended to other IPs IMO

Reply:
keeping DT_REG_ADDR_BY_IDX(DT_INST_PARENT(0), 1)